cli/src/semgrep/formatter/sarif.py

```diff
@@ -144,7 +196,10 @@ def _dataflow_trace_to_thread_flows_sarif(rule_match: RuleMatch) -> Any:
                 for intermediate_var_location in intermediate_var_locations:
                     locations.append(intermediate_var_location)
 
-        locations.append(SarifFormatter._sink_to_thread_flow_location_sarif(rule_match))
+        sink_thread_trace = SarifFormatter._rec_taint_obj_to_thread_flow_location_sarif(
```

Comment:
For the sink, there is no need to call a recursive function.
Please extract a non-recursive function that will be called here, and also used in `_rec_taint_obj_to_thread_flow_location_sarif`

Reply:
You convinced me that I misunderstood :-)